core/src/graph/mod.rs

```diff
@@ -186,6 +189,106 @@ derive_from! {
     }
 }
 
+impl Display for Value {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", serde_json::to_string(&self).unwrap())
```

Comment:
It might be faster to just match on self and forward to the relevant impls. `serde_json` has some machinery we likely don't need.